app/routes/lending/useFetchedLendableObject.ts

```diff
@@ -0,0 +1,38 @@
+import { usePreparedEffect } from '@webkom/react-prepare';
+import { useState } from 'react';
+import { fetchLendableObjectById } from 'app/actions/LendableObjectActions';
+import { selectLendableObjectById } from 'app/reducers/lendableObjects';
+import { useAppDispatch, useAppSelector } from 'app/store/hooks';
+import type { EntityId } from '@reduxjs/toolkit';
+import type {
+  DetailLendableObject,
+  UnknownLendableObject,
+} from 'app/store/models/LendableObject';
+
+export const useFetchedLendableObject = (lendableObjectId: EntityId) => {
+  const dispatch = useAppDispatch();
+  const [fetching, setFetching] = useState(true);
+
+  usePreparedEffect(
+    'fetchLendableObjectById',
+    () =>
+      dispatch(fetchLendableObjectById(lendableObjectId)).then(() =>
+        setFetching(false),
+      ),
+    [],
+  );
+
+  const lendableObject = useAppSelector((state) =>
+    selectLendableObjectById(state, lendableObjectId!),
+  );
+
+  return { lendableObject, fetching } as
+    | {
+        lendableObject?: UnknownLendableObject;
+        fetching: true;
+      }
+    | {
+        lendableObject: DetailLendableObject;
+        fetching: false;
+      };
```

Comment:
Can't this be handled by the reducer?

Reply:
Kind of, but the types will be wrong. The reducer simply returns:
```typescript
lendableObject?: UnknownLendableObject;
fetching: boolean;
```
which turns into:
```typescript
lendableObject?: DetailLendableObject;
fetching: boolean;
```
if you give the selector a type argument `<DetailLendableObject>`. Because of this it will assume that we have a `DetailLendableObject` whenever `lendableObject !== undefined`. However, we might still be fetching the detailed object and only have a `ListLendableObject` in the redux state. This could cause errors if we access properties that only the detailed serializer includes.

These types worked out very well when used on the page, so I am trying ot figure out a good way of generalizing this to all selectors. Hopefully you will see a PR on that soon:)